### DISCUSSION TITLE
SUGGESTION: Encapsule player production in a player factory

### DISCUSSION THREAD

**Discussion by Woernchen** [voxeltycoon/issues#1018] [General]
I like the idea of the researchable production lines (conveyor belt,...). 
This concept mixes macro management, like move good from a to b with the micro management, like produce boards out of wood logs up. Over the time I play this game, I have more and more problems with it. 
Why? For my taste the mirco management blocks to much space on the macro level. 
For example the chain saw + 2 depots + conveyor belts + connector uses more space than a normal factory on the macro level.

**Maybe it would be nice to hide te micro management inside a player factory on the global map.** 

The player would setup a factory, which looks like any other factory on the map. The difference is, that the player can open the factory and has then some area to place the storages, conveyor belts and machinery. This would separate both concepts a littlebit, but still keep the spirit. 
One could come up with several sizes of factories, either fixed or depending on the used space on the macro level. The interface between both levels would be the storage of the factory.

**Comment by karyonite** [comment]
Interesting thought. Do you feel that there is not enough space in the world to house the hand-made factories?

**Reply by Woernchen** [reply]
I guess there is enough space. Maybe I have a different playstyle. 
To be honest I am playing the game, because it is a grand child of transport tycoon. I also understand, that is at the same time child of factorio. So I am a little bit afraid of having a ball of production lines in the center of the map as a "jack of all trades" hand made factory. But maybe I am wrong, who knows.

Usually I am trying to reduce the amount of transportation routes as far as possible. 
In my current game I had a city which needed coal and iron, so I build my steel production next to it. Resulting in a production line surrounded by the city. When I want to expand it, one of them needs to move.
![VoxelTycoon_example](https://user-images.githubusercontent.com/83179520/218059539-4719ed81-08a2-4c1d-8c4b-85d22df5ae93.jpg)

**Reply by karyonite** [reply]
If I understand correctly, you are afraid that you’ll just build one massive factory, without a huge need to move items around, is that corrext? So actually… having enough space is part of the problem then? 

Although I see how limiting the space inside a fixed ‘factory’ building could add some logical constraints, what would prevent us from just plopping several of those factories together? To me it seems the challenge still remains, just adding more hoops to jump through.

However! Just thinking out loud… what if every town/municipality had a limited sized zone to allow for custom made industrial buildings? With some towns not even allowing factory construction at all?

**Reply by Woernchen** [reply]
I guess, it is not all about space. Also I clearly understand, why you are asking and asking. 

One challenge I see is the mixture of both concepts (macro vs. micro) on one map. 
The other is that the sizing of player buildings is different compared to the one from the AI. For me it feels unbalanced. 
For Example one AI-Factory has the size of a medium depot. For a simple production line one need already 2 depots, the machinery, connectors, and conveyour belts.     

Just some heretic thoughts I had: Why do you need trains and trucks at all? One could use only depots and conveyor belts. This would have the advantage of having a continous stream of goods. 

However I like your idea of having industrial areas per town. 
-> Maybe the amount could be linked to the level of the town.

**Comment by karyonite** [comment]
Thanks for indulging me :) I'm just a player like you and I want to really understand the problem you're experiencing. I also think it will help the developers to see several sides of a problem being discussed and chewed on, because solving one problem usually causes 10 others ;D

And I guess your thought is not as heretic as you might think. Factorio does supply and demand with belts, trains and drones. Satisfactory adds cars to the roster, that can drive freely from one depot to another. Giving the players options on how to play, is part of the charm, to me. 

However, since Voxel Tycoon seems to lean heavily on the TTD-style logistics game, having map-spanning factories would take away from the game it's trying to be. In factory building games, you **are** the one changing the world. In logistics games, you **help** the world to change and grow. At least that's how I see it, I want to see how the world (towns) develop because of my input. (This is basically the point of my post about making the world feel more alive).
